RecoLocalCalo/EcalRecProducers/plugins/alpaka/EcalMultifitParametersHostESProducer.cc

```diff
@@ -0,0 +1,94 @@
+#include <cassert>
+
+#include "FWCore/ParameterSet/interface/ParameterSet.h"
+
+#include "CondFormats/EcalObjects/interface/alpaka/EcalMultifitParametersDevice.h"
+#include "CondFormats/EcalObjects/interface/EcalMultifitParametersSoA.h"
+#include "CondFormats/DataRecord/interface/EcalMultifitParametersRcd.h"
+
+#include "DataFormats/EcalDigi/interface/EcalConstants.h"
+
+#include "HeterogeneousCore/AlpakaCore/interface/alpaka/ESGetToken.h"
+#include "HeterogeneousCore/AlpakaCore/interface/alpaka/ESProducer.h"
+#include "HeterogeneousCore/AlpakaCore/interface/alpaka/ModuleFactory.h"
+#include "HeterogeneousCore/AlpakaInterface/interface/config.h"
+#include "HeterogeneousCore/AlpakaInterface/interface/host.h"
+#include "HeterogeneousCore/AlpakaInterface/interface/memory.h"
+
+namespace ALPAKA_ACCELERATOR_NAMESPACE {
+  class EcalMultifitParametersHostESProducer : public ESProducer {
+  public:
+    EcalMultifitParametersHostESProducer(edm::ParameterSet const&);
+    ~EcalMultifitParametersHostESProducer() override = default;
+
+    static void fillDescriptions(edm::ConfigurationDescriptions&);
+    std::unique_ptr<EcalMultifitParametersHost> produce(EcalMultifitParametersRcd const&);
+
+  private:
+    edm::ParameterSet const pset_;
```

Comment:
@makortel is it OK to make a copy of the `ParameterSet`, or is it better to extract the four `std::vector<double>` parameters here and make a copy of those, instead ?

Reply:
It is better to extract the parameters and store them in as compact form as possible in the ESProducer member data.

The `ParameterSet` takes always more memory, and getting anything from it is expensive-enough to care about it (even if the parsing would be limited to IOV changes here).